### ISSUE TITLE
[FEATURE] Support in-cluster PXE booting service to orchestrate mass deployment

### ISSUE THREAD

**Issue by innobead** [harvester/harvester#491]
**Is your feature request related to a problem? Please describe.**

As per #217 , PXE boot is supported in the Harvester installer, but users still need to set up the required environment for the complete deployment like iPXE/PXE server. 

For better workflow for the user experience, we should provide a built-in booting service (ex: PXE/iPXE) for the complete automatic node deployment and provisions like the features provided by Metal3 bare mental operator or Tinkerbell boots service, and Canonical MAAS provisioner, then users are able to manage via APIs and UI (user interfaces).

**Describe the solution you'd like**

- A booting server is required to set up when the cluster (the first control plane node) setup ready
- The related APIs are supported to configure nodes ready for deployment and provisioning
- The remaining nodes will be deployed and joined to the cluster
- For other tech details, an RFC is required

**Describe alternatives you've considered**
#217 supported only, and users need to set up the infra for running PXE booting flow.

**Additional context**
- https://github.com/tinkerbell/boots
- https://github.com/metal3-io/baremetal-operator

cc @bk201 @wjimenez5271

**Comment by evrardjp** [comment]
Yeah alternatively, I suppose that any IPMI/redfish would be useful?

**Comment by yasker** [comment]
@ibrokethecloud is working on a POC based on Tinkerbell.

Just a thought about the basic UX:
1. Install the first node using ISO.
    1. Unattended installation via ISO is needed to make this first step automated.
2. Once Harvester is installed and booted up, the user can now open UI, enable the component `Automated Mass Deployment`(temporary name) feature, which will deploy Tinkerbell via Helm chart.
    1. Alternatively, the user can call the API to make the change.
3. New API and UI will be available for the user to set the mac addresses (and other information e.g. hostname) of the new nodes that wanted to boot up and join the cluster. 
4. The other nodes will boot up then automatically install, reboot, and join the cluster.
5. User can disable the component, which will remove the Tinkerbell from the system.

On the Rancher side, we are also considering some other ways e.g. use Tinkerbell CAPI for provisioning.

**Comment by Yasumoto** [comment]
This would be very exciting! @ibrokethecloud I searched around a bit, looks like this is where you’re at? 🚀 

https://github.com/ibrokethecloud/harvester-tink-operator

I think I’ll kick the tires here (we’re also looking at Harvester overall, thanks for all the hard work!) and see what I run into!

**Comment by yasker** [comment]
Related to https://github.com/harvester/harvester/issues/2346 .

Seeder is up at https://github.com/harvester/seeder
